src/NHibernate/Dialect/MsSql2008Dialect.cs

```diff
@@ -36,5 +74,26 @@ protected override void RegisterDefaultProperties()
 			base.RegisterDefaultProperties();
 			DefaultProperties[Environment.ConnectionDriver] = typeof(Sql2008ClientDriver).AssemblyQualifiedName;
 		}
+
+		public override string CurrentTimestampSQLFunctionName =>
+			"SYSDATETIME()";
```

Comment:
Previously it was using [`CURRENT_TIMESTAMP`](https://docs.microsoft.com/en-us/sql/t-sql/functions/current-timestamp-transact-sql), calling indeed `getdate()`, so yielding `datetime` instead of `datetime2`.